pkg/ds/queue/queue.go

```diff
@@ -0,0 +1,155 @@
+package queue
+
+type Node[T any] struct {
+	value T
+	next  *Node[T]
+	prev  *Node[T]
+}
+
+type Queue[T any] struct {
+	head *Node[T]
+	tail *Node[T]
+	size uint
+}
+
+func New[T any]() *Queue[T] {
+	return &Queue[T]{
+		head: nil,
+		tail: nil,
+		size: 0,
+	}
+}
+
+func (q *Queue[T]) PushBack(val T) {
+	q.size++
+	node := &Node[T]{
+		value: val,
+		next:  nil,
+		prev:  nil,
+	}
+
+	if q.head == nil {
```

Comment:
I think it's better to check `q.IsEmpty` here, need to modify code for doing so.

Reply:
updated in https://github.com/KyberNetwork/tradinglib/pull/13/commits/b77b7dc94ee7c6d4ae2eeaf97ac3686310d30d5e